### PR TITLE
Artwork Size illustration

### PR DESCRIPTION
**Pull request by stkeogie**
Added Artwork size illustration ( #299 )
Here are few examples:
artwork/Q3862632
artwork/Q18573716
artwork/Q3937687

illustration is only shown if the artwork is larger than 9cm x 9cm and smaller than 35m.
Here are examples that must not be displayed.
artwork/Q60775779
artwork/Q56397959

### CODE REVIEW

**Comment by BernhardHumm** [comment]
As discussed today:

 

Possible improvements / extensions:

 

Artwork up to the size of a human: fixed size of human figure; (smaller) artwork aligned to the head of the human.

Artworks taller than a human: fixed size of artwork; (smaller) human aligned to the bottom of the artwork

 

 

 

------------------------------------

 

 

Von: stkeogie [mailto:notifications@github.com] 
Gesendet: Donnerstag, 28. Mai 2020 12:55
An: hochschule-darmstadt/openartbrowser <openartbrowser@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Betreff: [hochschule-darmstadt/openartbrowser] Artwork Size illustration (#365)

 

Added Artwork size illustration ( #299 <https://github.com/hochschule-darmstadt/openartbrowser/issues/299>  )
Here are few examples:
artwork/Q3862632
artwork/Q18573716
artwork/Q3937687

illustration is only shown if the artwork is larger than 9cm x 9cm and smaller than 35m.
Here are examples that must not be displayed.
artwork/Q60775779
artwork/Q56397959

  _____  


You can view, comment on, or merge this pull request online at:


  https://github.com/hochschule-darmstadt/openartbrowser/pull/365


Commit Summary


*	Displaying Artwork size-impression, when height & width are given & units are in cm
*	Added more supported illustration units. Disabled Illustration when artwork is to small or large.
*	Aligned Illustration at top when artwork is small. Added Height Illustration when artwork is greater than 5.5m and less 35m
*	Hide Dimension illustration if image is smaller than 9cm x 9cm


File Changes


*	M app/src/app/shared/components/dimensions/dimensions.component.html <https://github.com/hochschule-darmstadt/openartbrowser/pull/365/files#diff-622503b42fbb2fccded28e9dc149fe7f>  (93) 
*	M app/src/app/shared/components/dimensions/dimensions.component.scss <https://github.com/hochschule-darmstadt/openartbrowser/pull/365/files#diff-f3e0f6fc54c3bdabef5e1774b6ade6d3>  (12) 
*	M app/src/app/shared/components/dimensions/dimensions.component.ts <https://github.com/hochschule-darmstadt/openartbrowser/pull/365/files#diff-a7eaaabf470c1d09cb8a18f9a6f3c6e5>  (34) 


Patch Links:


*	https://github.com/hochschule-darmstadt/openartbrowser/pull/365.patch
*	https://github.com/hochschule-darmstadt/openartbrowser/pull/365.diff

File: app/src/app/shared/components/dimensions/dimensions.component.html [hunk 1]
```diff
@@ -1,26 +1,75 @@
...
+    [ngClass]="{ 'illustration-margin': illustrationHeight < 92 }"
+    xml:space="preserve"
+    [attr.width.px]="illustrationWidth"
+    [attr.height.px]="illustrationHeight"
+  >
+    <rect width="calc(100% - 0.5px)" height="calc(100% - 0.5px)" style="fill:#888;stroke:white;stroke-width:2;opacity:0.5" />
+  </svg>
+</div>
+<!-- If artwork height is greater than 5.5m and smaller than 35 m, illustrate height -->
+<div
```
Comment (EeveesEyes) [inline app/src/app/shared/components/dimensions/dimensions.component.html:47]: This looks like code-duplication. Is this really necessary?

File: app/src/app/shared/components/dimensions/dimensions.component.ts [hunk 2]
```diff
@@ -19,14 +19,46 @@ export class DimensionsComponent implements OnInit {
...
 
   ngOnInit() {
     if (this.artwork) {
       this.setDimensions();
+      this.calculateIllustrationDimensions();
+    }
+  }
+  calculateIllustrationDimensions() {
+    const scalingFactor = 0.53;
+    if (this.artwork.height) {
```
Comment (EeveesEyes) [inline app/src/app/shared/components/dimensions/dimensions.component.ts:37]: This looks like code-duplication. Please move the switch in a private function.

File: app/src/app/shared/components/dimensions/dimensions.component.html [hunk 3]
```diff
@@ -1,26 +1,75 @@
...
+    </g>
+  </svg>
+
+  <svg
+    [ngClass]="{ 'illustration-margin': illustrationHeight < 92 }"
+    xml:space="preserve"
+    [attr.width.px]="illustrationWidth"
+    [attr.height.px]="illustrationHeight"
+  >
+    <rect width="calc(100% - 0.5px)" height="calc(100% - 0.5px)" style="fill:#888;stroke:white;stroke-width:2;opacity:0.5" />
```
Comment (EeveesEyes) [inline app/src/app/shared/components/dimensions/dimensions.component.html:43]: Please move style attributes to .scss file.

**Comment by BernhardHumm** [comment]
I agree: It’s a great feature and swapping even improves it.

 

 

------------------------------------

 

Von: Adrian Kailus [mailto:notifications@github.com] 
Gesendet: Freitag, 29. Mai 2020 13:14
An: hochschule-darmstadt/openartbrowser <openartbrowser@noreply.github.com>
Cc: Humm, Bernhard, Prof. Dr. <bernhard.humm@h-da.de>; Comment <comment@noreply.github.com>
Betreff: Re: [hochschule-darmstadt/openartbrowser] Artwork Size illustration (#365)

 

@EeveesEyes requested changes on this pull request.

I love this feature, it looks awesome. One suggestion though:

What about moving it directly above the dimension like this:
 <https://user-images.githubusercontent.com/28682931/83253583-7dc99b00-a1ad-11ea-9730-f42a5bf5e8f9.png> 

  _____  

In app/src/app/shared/components/dimensions/dimensions.component.html <https://github.com/hochschule-darmstadt/openartbrowser/pull/365#discussion_r432415139> :

> +        d="m62.096 8.5859c-5.208 0-9.424 4.2191-9.424 9.4261 0.001 5.203 4.217 9.424 9.424 9.424 5.202 0 9.422-4.221 9.422-9.424 0-5.208-4.22-9.4261-9.422-9.4261zm-10.41 21.268c-6.672 0-12.131 5.407-12.131 12.07v29.23c0 2.275 1.791 4.123 4.07 4.123 2.28 0 4.127-1.846 4.127-4.123v-26.355h2.102s0.048 68.811 0.048 73.331c0 3.05 2.478 5.53 5.532 5.53 3.052 0 5.525-2.48 5.525-5.53v-42.581h2.27v42.581c0 3.05 2.473 5.53 5.531 5.53 3.054 0 5.549-2.48 5.549-5.53v-73.331h2.127v26.355c0 2.275 1.85 4.123 4.126 4.123 2.28 0 4.073-1.846 4.073-4.123v-29.23c0-6.663-5.463-12.07-12.129-12.07h-20.82z"
+      />
+    </g>
+  </svg>
+
+  <svg
+    [ngClass]="{ 'illustration-margin': illustrationHeight < 92 }"
+    xml:space="preserve"
+    [attr.width.px]="illustrationWidth"
+    [attr.height.px]="illustrationHeight"
+  >
+    <rect width="calc(100% - 0.5px)" height="calc(100% - 0.5px)" style="fill:#888;stroke:white;stroke-width:2;opacity:0.5" />
+  </svg>
+</div>
+
+<div

This looks like code-duplication. Is this really necessary?

  _____  

In app/src/app/shared/components/dimensions/dimensions.component.ts <https://github.com/hochschule-darmstadt/openartbrowser/pull/365#discussion_r432415547> :

>    constructor() { }
 
   ngOnInit() {
     if (this.artwork) {
       this.setDimensions();
+      this.calculateIllustrationDimensions();
+    }
+  }
+  calculateIllustrationDimensions() {
+    const scalingFactor = 0.53;
+    if (this.artwork.height) {

This looks like code-duplication. Please move the switch in a private function.

  _____  

In app/src/app/shared/components/dimensions/dimensions.component.html <https://github.com/hochschule-darmstadt/openartbrowser/pull/365#discussion_r432416088> :

> +    <g ns:extraneous="self">
+      <path
+        style="fill: #888; stroke: black;"
+        stroke-width="0.8"
+        d="m62.096 8.5859c-5.208 0-9.424 4.2191-9.424 9.4261 0.001 5.203 4.217 9.424 9.424 9.424 5.202 0 9.422-4.221 9.422-9.424 0-5.208-4.22-9.4261-9.422-9.4261zm-10.41 21.268c-6.672 0-12.131 5.407-12.131 12.07v29.23c0 2.275 1.791 4.123 4.07 4.123 2.28 0 4.127-1.846 4.127-4.123v-26.355h2.102s0.048 68.811 0.048 73.331c0 3.05 2.478 5.53 5.532 5.53 3.052 0 5.525-2.48 5.525-5.53v-42.581h2.27v42.581c0 3.05 2.473 5.53 5.531 5.53 3.054 0 5.549-2.48 5.549-5.53v-73.331h2.127v26.355c0 2.275 1.85 4.123 4.126 4.123 2.28 0 4.073-1.846 4.073-4.123v-29.23c0-6.663-5.463-12.07-12.129-12.07h-20.82z"
+      />
+    </g>
+  </svg>
+
+  <svg
+    [ngClass]="{ 'illustration-margin': illustrationHeight < 92 }"
+    xml:space="preserve"
+    [attr.width.px]="illustrationWidth"
+    [attr.height.px]="illustrationHeight"
+  >
+    <rect width="calc(100% - 0.5px)" height="calc(100% - 0.5px)" style="fill:#888;stroke:white;stroke-width:2;opacity:0.5" />

Please move style attributes to .scss file.

**Comment by mauamy** [comment]
SelfPraiseMeme

**Pull request merged**